### PR TITLE
add test to ensure there's only one variant for each dependency in dep graph of easyconfigs using particular toolchains (REVIEW)

### PR DESCRIPTION
**Pull request by boegel**

### CODE REVIEW

**Comment by boegel** [comment]
I propose to fix the remaining violations that cause the newly added test to fail via changes implemented in #5993 and #5994 .

**Comment by vanzod** [comment]
Re-trigger Travis

**Comment by boegel** [comment]
@easybuilders/easybuild-easyconfigs-maintainers With #5993 and #5994 merged, this seems good to go...

File: test/easyconfigs/easyconfigs.py [hunk 1]
```diff
@@ -151,6 +153,88 @@ def test_conflicts(self):
...
+                # for Python & Tkinter, it's OK to have on 2.x and one 3.x version
+                v2_dep_vars = [x for x in dep_vars.keys() if x.startswith('version: 2.')]
+                v3_dep_vars = [x for x in dep_vars.keys() if x.startswith('version: 3.')]
+                if len(v2_dep_vars) == 1 and len(v3_dep_vars) == 1:
+                    res = True
+
+            return res
+
+        # restrict to checking dependencies of easyconfigs using common toolchains (start with 2018a)
+        for pattern in ['201[89][ab]', '20[29][0-9][ab]']:
```
Comment (migueldiascosta) [inline test/easyconfigs/easyconfigs.py]: shouldn't the second pattern be `20[2-9][0-9][ab]`?

File: test/easyconfigs/easyconfigs.py [hunk 2]
```diff
@@ -151,6 +153,88 @@ def test_conflicts(self):
...
+                empty_vsuff_vars = [v for v in dep_vars.keys() if v.endswith('versionsuffix: ')]
+                if len(empty_vsuff_vars) == 1:
+                    dep_vars = [(k, v) for (k, v) in dep_vars.items() if k != empty_vsuff_vars[0]]
+
+            # only single variant is always OK
+            if len(dep_vars) == 1:
+                res = True
+
+            elif len(dep_vars) == 2 and dep in ['Python', 'Tkinter']:
+                # for Python & Tkinter, it's OK to have on 2.x and one 3.x version
```
Comment (migueldiascosta) [inline test/easyconfigs/easyconfigs.py]: `s/on 2/one 2/`

**Comment by migueldiascosta** [comment]
Going in, thanks @boegel!

**Comment by boegel** [comment]
🎉

**Pull request merged**